Platform/UTMJailbreak.m

```diff
@@ -56,15 +82,99 @@ static bool am_i_being_debugged() {
     return !csops(getpid(), CS_OPS_STATUS, &flags, sizeof(flags)) && flags & CS_DEBUGGED;
 }
 
+static NSDictionary *parse_entitlements(const void *entitlements, size_t length) {
+    char *copy = malloc(length);
+    memcpy(copy, entitlements, length);
+    
+    // strip out psychic paper entitlement hiding
+    if (@available(iOS 13.5, *)) {
+    } else {
+        static const char *needle = "<!---><!-->";
+        char *found = strnstr(copy, needle, length);
+        if (found) {
+            memset(found, ' ', strlen(needle));
+        }
+    }
+    NSData *data = [NSData dataWithBytes:copy length:length];
+    free(copy);
+    
+    return [NSPropertyListSerialization propertyListWithData:data
+                                                     options:NSPropertyListImmutable
+                                                      format:nil
+                                                       error:nil];
+}
+
+static NSDictionary *app_entitlements(void) {
+    static NSDictionary *entitlements;
+    if (entitlements != nil)
+        return entitlements;
+    
+    // Inspired by codesign.c in Darwin sources for Security.framework
+    
+    // Find our mach-o header
+    Dl_info dl_info;
+    if (dladdr(app_entitlements, &dl_info) == 0)
+        return nil;
+    if (dl_info.dli_fbase == NULL)
+        return nil;
+    char *base = dl_info.dli_fbase;
+    struct mach_header_64 *header = dl_info.dli_fbase;
```

Comment:
Consider

```
#include <mach-o/ldsyms.h>

struct mach_header_64 *header = &_mh_execute_header;
```

Reply:
I got the code from https://github.com/ish-app/ish/blob/master/app/AppGroup.m I don't want to make too much changes, since I want to easily copy-paste changes from iSH if needed in the future. Also, it seems like we need `dl_info` anyways for `base`, so I don't think we would gain much here. And since this extern variable starts with an underscore, I'm afraid that it might get changed or removed in the future without warning.

Reply:
FWIW:

<img width="363" alt="Screen Shot 2020-09-20 at 15 45 03" src="https://user-images.githubusercontent.com/13786931/93723998-4154ed00-fb58-11ea-92ea-35878ce568e3.png">

Is `base` just not the address of the thing? (Also, there's `_NSGetMachExecuteHeader` from <crt_externs.h>, if that makes you more confident in this continuing to exist.)

Reply:
I’ll look into that.